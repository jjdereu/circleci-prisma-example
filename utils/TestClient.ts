import * as http from 'http'
import * as https from 'https'
import { GraphQLClient } from 'graphql-request'
const debug = require('debug')('TestClient')
import { GraphQLServer } from 'graphql-yoga'
import { server } from '../src'

export default class TestClient {
  app: GraphQLServer
  server: http.Server
  address: any
  client: GraphQLClient
  initPromise: Promise<void>
  constructor() {
    this.app = server
    this.initPromise = this.init()
    debug({ address: this.address })
  }
  async init(): Promise<void> {
    if (!this.server) {
      this.server = (await this.app.start({
        port: 0,
        debug: false,
      })) as any
    }
    const { port }: any = this.server.address()
    const protocol = this.server instanceof https.Server ? 'https' : 'http'
    this.address = `${protocol}://127.0.0.1:${port}/`
    this.client = new GraphQLClient(this.address)
  }
  async request<T = any>(
    query: string,
    variables?: any,
    headers?: any,
  ): Promise<any> {
    debug(this.address, query, variables)
    await this.initPromise
    if (headers) {
      return new GraphQLClient(this.address, { headers }).request(
        query,
        variables,
      )
    }
    return this.client.request<T>(query, variables)
  }
}