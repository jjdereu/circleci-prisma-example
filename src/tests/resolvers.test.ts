import TestClient from '../../utils/TestClient'

const client = new TestClient()

describe('Query and mutation work', () => {
  it('should create a draft', async () => {
    const res: any = await client.request(`
      mutation {
        createDraft(title: "Title", content: "Content", authorEmail: "alice@prisma.io") {
          title
        }
      }
    `)
    expect(res).toMatchSnapshot()
  })

  it('should receive the test draft', async () => {
    const res: any = await client.request(`
      {
        drafts {
          title
        }
      }
    `)
    expect(res.drafts[0]).toMatchSnapshot()
  })

  it('should delete the test draft', async () => {
    const res: any = (parent, args, ctx) => ctx.db.deleteManyPosts({ where: {title: "Title"}})
    expect(res).toBeDefined()
  })
})

