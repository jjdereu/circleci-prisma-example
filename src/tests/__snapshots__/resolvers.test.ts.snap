// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query and mutation work should create a draft 1`] = `
Object {
  "createDraft": Object {
    "title": "Title",
  },
}
`;

exports[`Query and mutation work should receive the test draft 1`] = `
Object {
  "title": "Title",
}
`;
